pkg/controller/controller.go

```diff
@@ -739,6 +780,65 @@ func findServicePlan(name string, plans []v1alpha1.ServicePlan) *v1alpha1.Servic
 	return nil
 }
 
+// handleLastOperationResponse gets called when a last operation poll has happened.
+func (c *controller) handleLastOperationResponse(instance *v1alpha1.Instance, resp *brokerapi.LastOperationResponse) bool {
+	switch resp.State {
+	case "in progress":
+		c.setAsyncOperationOngoing(instance, true)
+		return true
+	case "succeeded":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to succeed
+	case "failed":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to fail
+	default:
+		glog.Warningf("Got invalid state in LastOperationResponse: %q", resp.State)
+	}
+	return false
+}
+
+// setAsyncOperationOngoing updates the
+// InstanceConditionAsyncOperationInProgress to the specified status.
+func (c *controller) setAsyncOperationOngoing(instance *v1alpha1.Instance, status bool) {
+	if status {
+		c.updateInstanceCondition(
```

Comment:
and then we need AsyncUpdateInProgress, AsyncDeprovisionInProgress and since the bindings also are probably going to have that we'll basically be doing ~5 string compares. As I said, I think that's janky but whatever.